packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts

```diff
@@ -0,0 +1,277 @@
+  logShowMoreFoldedResults: (result: Result) => ClickAction;
+  logShowLessFoldedResults: () => CustomAction;
+}
+
+/**
+ * Creates a core `FoldedResultList` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `FoldedResultList` properties.
+ * @param analyticsClient - A FoldedResultAnalyticsClient to send the appropriate analytics calls.
+ * @returns A `CoreFoldedResultList` controller instance.
+ */
+export function buildCoreFoldedResultList(
+  engine: CoreEngine,
+  props: CoreFoldedResultListProps,
+  analyticsClient: FoldedResultAnalyticsClient
+): CoreFoldedResultList {
+  if (!loadFoldingReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildCoreResultList(engine, props);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  const options = props?.options?.folding
+    ? validateOptions(
+        engine,
+        optionsSchema,
+        props.options!.folding!,
+        'buildFoldedResultList'
+      )
+    : {};
+
+  dispatch(registerFolding({...options}));
+
+  return {
+    ...controller,
+
+    loadCollection: (collection) => {
+      dispatch(
+        props.loadCollectionActionCreator(
+          collection?.result?.raw[
+            engine.state.folding.fields.collection
+          ] as string
+        )
+      );
+      dispatch(analyticsClient.logShowMoreFoldedResults(collection?.result));
+    },
+    logShowMoreFoldedResults: (result) => {
+      dispatch(analyticsClient.logShowMoreFoldedResults(result));
+    },
+    logShowLessFoldedResults: () => {
+      dispatch(analyticsClient.logShowLessFoldedResults());
+    },
+
+    findResultById(collection) {
+      return searchForResult(
+        this.state.results,
+        (r) => r.result.uniqueId === collection.result.uniqueId
+      );
+    },
+
+    findResultByCollection(collection) {
+      return searchForResult(
+        this.state.results,
+        (r) =>
+          r.result.raw.foldingcollection ===
+          collection.result.raw.foldingcollection
+      );
+    },
+
+    get state() {
+      const state = getState();
+
+      return {
+        ...controller.state,
+        results: controller.state.results.map((result) => {
+          const collectionId = result.raw[state.folding.fields.collection] as
+            | string
+            | undefined;
+          if (!collectionId || !state.folding.collections[collectionId]) {
+            return {
+              result,
+              moreResultsAvailable: false,
+              isLoadingMoreResults: false,
+              children: [],
+            };
+          }
+          return state.folding.collections[collectionId];
+        }),
+      };
+    },
+  };
+}
+
+function loadFoldingReducer(
+  engine: CoreEngine
+): engine is CoreEngine<
+  SearchSection & ConfigurationSection & FoldingSection & QuerySection
+> {
+  engine.addReducers({search, configuration, folding, query});
```

Comment:
Are all of these reducers needed for the insight use case?

Reply:
for the reducers I have used the same ones previous in use by the `headless-folded-result-list` controller.
It seems to work fine without the query reducer but I believe the 3 other ones to be necessary

Reply:
I'm a bit suspicious of the search reducer. Does the insight use case typically use the search reducer?

Reply:
I dont think the insight usecase uses it. I tested it on my end, both insight and search use cases seem to work even without the search reducer.
I would have thought it would be needed since the core controller contains both the search and insight usecases and the previous "search only" iteration of the controller seemed to add it to its engine

Reply:
I dug a bit more into this, turns out you were right search doesnt seem to be needed. `query`, however does.